Documentation/ApiOverview/FlexForms/Index.rst

```diff
@@ -0,0 +1,317 @@
+                        </TCEforms>
+                        <type>array</type>
+                        <el>
+                            <!-- Add settings here ... -->
+
+                            <!-- Example setting: input field with name settings.timeRestriction -->
+                            <settings.timeRestriction>
+                                <TCEforms>
+                                    <label>LLL:EXT:example/Resources/Private/Language/Backend.xlf:settings.registration.timeRestriction</label>
+                                    <config>
+                                        <default></default>
+                                        <type>input</type>
+                                        <size>15</size>
+                                    </config>
+                                </TCEforms>
+                            </settings.timeRestriction>
+
+                            <!-- end of settings -->
+
+                        </el>
+                    </ROOT>
+                </sDEF>
+            </sheets>
+        </T3DataStructure>
+
+
+#. The configuration schema is attached to one or more plugins
+
+   The vendor name is **Myvendor**, the extension key is **example**
+   and the plugin name is **Registration**.
+
+   In :file:`Configuration/TCA/Overrides/tt_content.php` add the following:
+
+   .. code-block:: php
+
+       $GLOBALS['TCA']['tt_content']['types']['list']['subtypes_addlist']['example_registration'] = 'pi_flexform';
+       \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPiFlexFormValue(
+           // plugin signature: <extension key> '_' <plugin name in lowercase>
+           'example_registration',
+           // Flexform configuration schema file
+           'FILE:EXT:example/Configuration/FlexForms/Registration.xml'
+       );
+
+   .. tip::
+
+      The plugin signature is used in the database field `tt_content.list_type`
+      as well, when the tt_content record is saved. If you are confused about
+      how to handle underscores and upper / lowercase, check there to see
+      what your plugin signature is.
+
+      Also look on the page :ref:`extension-naming`.
+
+#. The settings can be read using :php:`$this->settings` in an
+   Extbase controller.
+
+   .. code-block:: php
+
+      $includeCategories = (bool) ($this->settings['includeCategories'] ?? false);
+
+
+.. todo: Add information about how to read settings apart from Extbase Controllers.
+
+
+More Schema Examples
+====================
+
+The definition of the data types and parameters used complies in large
+parts to the :ref:`column types defined by TCA <t3tca:columns-types>`.
+
+The settings must be added within the <el> element in the FlexForm configuration
+schema file.
+
+.. important::
+
+    If you wish to access a setting from your controller via
+    :php:`$this->settings`, the name of the setting must begin with
+    **settings** directly followed by a dot (`.`).
+
+
+Select Field
```

Comment:
Looks very much like a duplicate of `Select a Controller Action`. I would not add the two remaining examples here, but to TCA Reference:

* https://docs.typo3.org/m/typo3/reference-tca/master/en-us/ColumnsConfig/Type/Select.html#items We should add your XML example here to keep structural changes regarding select items in a single place.
* https://docs.typo3.org/m/typo3/reference-tca/master/en-us/ColumnsConfig/Type/Select.html#itemsprocfunc We should add your example PHP implementation here!

Reply:
The goal of the FlexForms chapter is now to provide introduction with a few simple examples. I would like to keep the examples there. You are right, the TCA should be used as reference.